### PR TITLE
Fix typing issues

### PR DESCRIPTION
**Pull request by adamwawrzynski**
## Description

Fix and refine typing.

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes.

- [x] `pre-commit run --all-files`

## Checklist:

- [ ] I have updated all related `pyproject.toml` files.
- [ ] I have updated lock files.
- [x] I have set an Assignee and Reviewers.
- [x] I have labeled the PR correctly.
- [ ] I have performed a self-review of my own code.
- [ ] I have covered my code and changes with unit tests.
- [x] I have annotated types as extensively as possible.
- [ ] I have commented my code, particularly in hard-to-understand areas.
- [ ] I have made corresponding changes to the documentation.
- [x] My changes generate no new warnings.
- [x] All code references and dependencies will work.

### CODE REVIEW

File: foxai/explainer/computer_vision/object_detection/types.py [hunk 1]
```diff
@@ -11,7 +11,7 @@
 class PredictionOutput:
     """Data class for model prediction output in YOLO style."""
 
-    bbox: List[int]
+    bbox: Tuple[int, int, int, int]
```
Comment (joannasendorek) [inline foxai/explainer/computer_vision/object_detection/types.py:14]: Haven't you just added BboxType? Below

File: foxai/visualizer.py [hunk 2]
```diff
@@ -303,14 +303,14 @@ def get_heatmap_bbox(
...
-    bbox: List[int],
+    bbox: BboxType,
     heatmap: torch.Tensor,
     img: np.ndarray,
 ) -> np.ndarray:
     """Draw heatmap in bounding box on image.
 
     Args:
-        bbox: List of coordinates for bounding box.
+        bbox: Tuple of coordinates for bounding box in format `[x1, x2, y1, y2]`.
```
Comment (joannasendorek) [inline foxai/visualizer.py:313]:
```suggestion
        bbox: Tuple of coordinates for bounding box in format `(x1, x2, y1, y2)`.
```

File: foxai/visualizer.py [hunk 3]
```diff
@@ -289,7 +289,7 @@ def get_heatmap_bbox(
 
     Args:
         heatmap: Heatmap to visualize.
-        bbox: Bounding box of detection.
+        bbox: Bounding box of detection in format `[x1, x2, y1, y2]`.
```
Comment (joannasendorek) [inline foxai/visualizer.py:292]:
```suggestion
        bbox: Bounding box of detection in format `(x1, x2, y1, y2)`.
```

**Review by joannasendorek — APPROVED** [review]
Looks much cleaner, nice! Some small comments only to be fixed.

File: foxai/visualizer.py [hunk 4]
```diff
@@ -360,22 +360,21 @@ def object_detection_visualization(
     Returns:
         Array of series of images with heatmap displayed on detection bounding boxes.
     """
+    bbox: Tuple[int, int, int, int]
```
[the remark this answers is not in the document]
Comment (adamwawrzynski) [reply foxai/visualizer.py]: Fixed: https://github.com/softwaremill/FoXAI/pull/101/commits/eb61017843653e0cfede6da428c13587a28eab7f.

File: foxai/visualizer.py (hunk 3 above)
Comment (adamwawrzynski) [reply foxai/visualizer.py]: Fixed: https://github.com/softwaremill/FoXAI/pull/101/commits/eb61017843653e0cfede6da428c13587a28eab7f.

File: foxai/visualizer.py (hunk 2 above)
Comment (adamwawrzynski) [reply foxai/visualizer.py]: Fixed: https://github.com/softwaremill/FoXAI/pull/101/commits/eb61017843653e0cfede6da428c13587a28eab7f.

File: foxai/explainer/computer_vision/object_detection/types.py (hunk 1 above)
Comment (adamwawrzynski) [reply foxai/explainer/computer_vision/object_detection/types.py]: Fixed: https://github.com/softwaremill/FoXAI/pull/101/commits/eb61017843653e0cfede6da428c13587a28eab7f.

**Pull request merged**